ConsoleBank/Portal/Program.cs

```diff
@@ -0,0 +1,219 @@
+            {
+                Console.Write("You entered the wrong item. Please try again: ");
+            }
+
+            return itemSelect;
+        }
+
+        private static void ShowException(Exception ex)
+        {
+            ConsoleColor color = Console.ForegroundColor;
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine(ex.Message);
+            Console.ForegroundColor = color;
+            Console.ReadKey();
+        }
+
+        private static void PrintMenu()
+        {
+            Console.Clear();
+            ConsoleColor color = Console.ForegroundColor;
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine("1. Print currency by id to the console");
+            Console.WriteLine("2. Print currency by abbreviation to the console");
+            Console.WriteLine("3. Print currency by id and date to the console");
+            Console.WriteLine("4. Print currency by abbreviation and date to the console");
+            Console.WriteLine("5. Convert to BYN");
+            Console.WriteLine("6. Convert from BYN");
+            Console.WriteLine("7. Exit program");
+
+            Console.WriteLine("Enter the item number:");
+            Console.ForegroundColor = color;
+        }
+
+        private static async Task PrintCurrencyById()
+        {
+            PrintCurrency(await _service.GetCurrencyRate(ReadId(), null));
+        }
+
+        private static async Task PrintCurrencyByAbbreviation()
+        {
+            PrintCurrency(await _service.GetCurrencyRate(ReadAbbreviation(), null));
+        }
+
+        private static async Task PrintCurrencyByIdAndDate()
+        {
+            PrintCurrency(await _service.GetCurrencyRate(ReadId(), ReadDate()));
+        }
+
+        private static async Task PrintCurrencyByAbbreviationAndDate()
+        {
+            PrintCurrency(await _service.GetCurrencyRate(ReadAbbreviation(), ReadDate()));
+        }
+
+        private static async Task ConvertToBYN()
+        {
+            var currencyId = await _service.GetCurrencyId(ReadAbbreviation());
+
+            PrintConvertMoney(await _convertor.ConvertToByn(currencyId, ReadAmount()), "BYN");
+        }
+
+        private static async Task ConvertFromBYN()
+        {
+            var abbreviation = ReadAbbreviation();
+            var currencyId = await _service.GetCurrencyId(abbreviation);
+
+            PrintConvertMoney(await _convertor.ConvertFromByn(currencyId, ReadAmount()), abbreviation);
+        }
+
+        private static void PrintCurrency(CurrencyRate rate)
+        {
+            Console.WriteLine(rate);
+        }
+
+        private static void PrintConvertMoney(double amount, string abbreviation)
+        {
+            Console.WriteLine($"{amount:0.0000} {abbreviation}");
+        }
+
+        private static void AddWhiteListForCurrenty()
+        {
+            _listCurrency.AddRange(new List<string> { "USD", "EUR", "RUB" });
```

Comment:
отдельный метод не сильно имеет смысл, можно в поле записать сразу
```suggestion
            _availableCurrencies.AddRange(new List<string> { "USD", "EUR", "RUB" });
```

Reply:
Метод для белого листа аббревиатур я оставил. Как по мне его лучше оставить. Так остается больше возможности для модернизации.